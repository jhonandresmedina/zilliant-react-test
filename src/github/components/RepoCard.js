import React from 'react';
import { Avatar, Card, CardTitle, CardText, List, ListItem } from 'react-md';

const RepoCard = ({ repo, selectRepo }) => {
    return (
        <Card onClick={() => selectRepo(repo.get('id'))}>
            <CardTitle
                title={repo.get('name')}
                subtitle={repo.get('full_name')}
                avatar={<Avatar random> {repo.get('name')[0].toUpperCase()} </Avatar>}
            />
            <CardText>
                <p>{repo.get('description')}</p>
                <List>
                    <ListItem
                        primaryText={repo.get('language') ? repo.get('language') : 'Unknown'}
                        secondaryText='Language'
                    />
                </List>
            </CardText>
        </Card>
    );
};

export default RepoCard;
