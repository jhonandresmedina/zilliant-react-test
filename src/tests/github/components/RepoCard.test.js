//@vendor
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { fromJS } from 'immutable';
import { Card } from 'react-md';

//@components
import RepoCard from '../../../github/components/RepoCard';

Enzyme.configure({
    adapter: new Adapter(),
});

const buildComponent = props => {
    let repo = fromJS({
        id: 77777,
        name: 'repo name',
        full_name: 'reactjs app',
        language: 'ES',
    });

    return <RepoCard repo={repo} {...props} />;
};

describe('RepoCard test', () => {
    it('renders correctly', () => {
        const wrapper = shallow(buildComponent());
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('click event', () => {
        const selectRepo = jest.fn();
        const wrapper = shallow(
            buildComponent({
                selectRepo,
            }),
        );
        const card = wrapper.find(Card);
        card.simulate('click');
        expect(selectRepo.mock.calls.length).toEqual(1);
    });
});
