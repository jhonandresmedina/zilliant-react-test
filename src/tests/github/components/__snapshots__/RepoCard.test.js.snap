// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard test renders correctly 1`] = `
<Card
  animate={true}
  expanderIcon={
    <FontIcon
      iconClassName="material-icons"
    >
      keyboard_arrow_down
    </FontIcon>
  }
  expanderTooltipPosition="left"
  onClick={[Function]}
>
  <CardTitle
    avatar={
      <Avatar
        random={true}
        suffixes={
          Array [
            "red",
            "pink",
            "purple",
            "deep-purple",
            "indigo",
            "blue",
            "light-blue",
            "cyan",
            "teal",
            "green",
            "light-green",
            "lime",
            "yellow",
            "amber",
            "orange",
            "deep-orange",
            "brown",
            "grey",
            "blue-grey",
          ]
        }
      >
         
        R
         
      </Avatar>
    }
    subtitle="reactjs app"
    title="repo name"
  />
  <CardText
    component="section"
  >
    <p />
    <List>
      <ListItem
        activeClassName="md-text--theme-primary"
        animateNestedItems={true}
        component="div"
        expanderIcon={
          <FontIcon
            iconClassName="material-icons"
          >
            keyboard_arrow_down
          </FontIcon>
        }
        itemComponent="li"
        primaryText="ES"
        renderChildrenOutside={false}
        secondaryText="Language"
      />
    </List>
  </CardText>
</Card>
`;
